pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -90,15 +90,12 @@ func (self *RefsHelper) CheckoutRef(ref string, options types.CheckoutRefOptions
 								if err := self.c.Git().Branch.Checkout(ref, cmdOptions); err != nil {
 									return err
 								}
-
-								onSuccess()
-								if err := self.c.Git().Stash.Pop(0); err != nil {
-									if err := self.c.Refresh(refreshOptions); err != nil {
-										return err
-									}
+								popErr := self.c.Git().Stash.Pop(0)
+								// Branch switch successful so re-render the UI even if the pop operation failed (e.g. conflict).
+								if err := renderBranchSwitched(); err != nil {
 									return err
 								}
-								return self.c.Refresh(refreshOptions)
+								return popErr
```

Comment:
If both the pop and the refresh fail, then you are returning the error from refresh. I think it would be better to return the error from pop instead. One way to do that (with a pattern that I have used elsewhere) is
```
		popErr := self.c.Git().Stash.Pop(0)
		// Branch switch successful so refresh the UI even if the pop operation failed (e.g. conflict).
		refreshError := renderBranchSwitched()
		if popErr != nil {
			// An error from pop is the more important one to report to the user
			return popErr
		}
		return refreshError
```

Alternatively, we could have renderBranchSwitched not return an error, and swallow the return value of the refresh call inside, knowing that a refresh with `BLOCK_UI` [will always return nil](https://github.com/jesseduffield/lazygit/blob/05ae0800c3ff3fd7160a5705a8fbac7f50745624/pkg/gui/controllers/helpers/refresh_helper.go#L203). Thinking about this, I guess I would prefer that as it results in simpler code.

Reply:
I made the error swallowing change for `CheckoutRef`. `NewBranch` had another source of error besides the refresh call (`self.c.Context().Push(self.c.Contexts().Branches)`) so for that one, I did the precedence switch approach.